src/Components/DynamicFields.res

```diff
@@ -117,7 +117,17 @@ let make = (
   React.useEffect0(() => {
     let clientTimeZone = Utils.dateTimeFormat(.).resolvedOptions(.).timeZone
     let clientCountry = Utils.getClientCountry(clientTimeZone)
-    setCountry(_ => clientCountry.countryName)
+    let initialCountry =
+      requiredFields
+      ->Utils.getDictFromJson
+      ->Utils.getDictFromObj("billing.address.country")
```

Comment:
Instead of using `billing.address.country` please make this check on AddressCountry